packages/extension-collaboration/src/collaboration.ts

```diff
@@ -173,6 +191,51 @@ export const Collaboration = Extension.create<CollaborationOptions>({
 
     const ySyncPluginInstance = ySyncPlugin(fragment, ySyncPluginOptions)
 
-    return [ySyncPluginInstance, yUndoPluginInstance]
+    return [
+      ySyncPluginInstance,
+      yUndoPluginInstance,
+      // Only add the filterInvalidContent plugin if content checking is enabled
+      this.editor.options.enableContentCheck
+      && new Plugin({
+        key: new PluginKey('filterInvalidContent'),
+        filterTransaction: tr => {
+          // Is this transaction from Yjs Sync Plugin?
+          if (isChangeOrigin(tr)) {
+
+            // When collaboration is disabled, prevent any sync transactions from being applied
+            if (this.storage.isCollaborationDisabled) {
+              // If collaboration is disabled, unregister the Yjs plugins to prevent further sync transactions
+              this.editor.unregisterPlugin(ySyncPluginKey)
+              this.editor.unregisterPlugin(yUndoPluginKey)
+              return false
+            }
+
+            // Did the doc actually change as a result of this transaction?
+            if (tr.docChanged) {
+
+              // Attempt to parse the content of the step
+              try {
+                const content = tr.doc.toJSON()
+
+                this.editor.schema.nodeFromJSON(content)
```

Comment:
Should also do a `content.check()` here to make the schema is valid in structure too